syntax = "proto3";

package sensor_data_messages;

// --- Enums (Foot Sensor Specific) ---
enum FootSensorMessageType {
  FOOT_SENSOR_UNKNOWN_MESSAGE_TYPE = 0; // Default value, should always be 0
  FOOT_SENSOR_DATA = 1;
}

// --- Foot Sensor Data Message ---
message FootSensorData {
  repeated uint32 readings = 2; // Array of 8 ADC readings
}

// Foot sensor specific message for logging session metadata
message FootSensorSensingData {
  string firmware_version = 1;
  uint32 sampling_frequency = 2;
  FootSensorMessageType message_type = 3; // Uses FootSensorMessageType
}

// Foot sensor specific message for marking the end of a logging session
message FootSensorSessionEnd {
  uint64 uptime_ms = 1;
}

// --- Main Wrapper Message for Foot Sensor logs ---
// Renamed to be unique for Foot Sensor logs
message FootSensorLogMessage {
  oneof payload {
    FootSensorData foot_sensor = 1;
    FootSensorSensingData sensing_data = 4; // Uses FootSensorSensingData
    FootSensorSessionEnd session_end = 5;  // Uses FootSensorSessionEnd
  }
}
